### PR TITLE
[Functions] Various v1 improvements

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description
Additional changes from review. See commit messages for a full list

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol:49]: Not sure that I love the addition of this helper function, but I see the notes [here](https://www.notion.so/Steve-s-Functions-Solidity-Review-Notes-64447db7e6a64ad9a5f0ce503186eef6?d=437f97af79e04d4dafd1198bfe9f0120&pvs=4#b5976ba516244674849b393626379002) and will refrain from blocking the review because of it.

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:94]: Maybe renaming `getEthSignedMessageHash` to `getPrefixedMessageHash` will make it more clear and resolve Steve's comment here: https://www.notion.so/chainlink/Steve-s-Functions-Solidity-Review-Notes-64447db7e6a64ad9a5f0ce503186eef6?pvs=4#1b476bcc44e14e71a7bfb55678f01e2b

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:94]: I'm going to combine them into one method in the next PR

File: contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol:49]: Do prefer how it was before with the duplicated code in the constructor? (I don't have a strong opinon here)

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: I like your thinking, I should clarify this change:
So I removed the check for the return value because:
The Router is the source of truth. It shouldn't be dependent on the Coordinator doing an action for it to continue. In the future where Coordinator contract might be DON owned, what if the Coordinator always returned false here? Then users could never get their blocked funds back.

**Review by bolekk — APPROVED** [review]
This looks good to me but please get at least one more approval before merging.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:431]: state changes here, some event should be emitted

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol
Comment (RensR) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: Who owns coordinator contracts? Are you implying the user cannot get funds back if the DON owned coordinator is malicious? That sounds out of scope

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (RensR) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:431]: I see that you've moved the event into the Subscription, but seeing as you're ignoring the return value of this function there is no way to tell if this actually deleted anything.

File: contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol:49]: nit: please don't place functions between the public and internal (_)updateConfig

Comment (RensR) [reply contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol:49]: I also don't love the third config setting function with a fairly confusing name.

File: contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol:100]: If this cannot be modified why isn't this immutable?

File: contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol:218]: using uint256 is cheaper as it casts to uint256 anyway and there is no downside of using the entire slot in memory as there is no packing in memory

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:5]: don't import ITypeAndVersion through another import

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:63]: nit: i prefer the 

  string public constant override typeAndVersion = "EVM2EVMOnRamp 1.0.0";

syntax over a "real" function

File: contracts/test/v0.8/functions/v0/Gas.test.ts
Comment (RensR) [inline contracts/test/v0.8/functions/v0/Gas.test.ts:45]: there are newer link tokens

File: contracts/test/v0.8/functions/v0/Gas.test.ts
Comment (RensR) [inline contracts/test/v0.8/functions/v0/Gas.test.ts:50]: there are newer MockV3Aggregator, generally never import outside of the v0.8 folder

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: This made me thing of another case where that we should make sure we think through carefully:
1. The DON attempts to call the user's callback, but fails for some reason that is not the fault of the user (ie: user doesn't get charged)
2. Since we delete the commitment, how can a user time out their request and have their funds unblocked?

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: In the above the user's funds will unblocked, because the Router handles subscriptions

Comment (bolekk) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: @KuphJr which row of the fulfillment events doc is that scenario in? We can discuss over there too.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:431]: I'll keep it in both places

File: contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/HasRouter.sol:49]: Rolling it back unless someone has a suggestion of how to fulfill Steve's point

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:531]: > Who owns coordinator contracts? Are you implying the user cannot get funds back if the DON owned coordinator is malicious? That sounds out of scope
Currently CLL, but yeah the hope is eventually DON owned.
The overall design was already heavily influenced by building toward this :shrug:

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:431]: Ended up adding an event here called `CommitmentDeleted` to be more consistent with the function name

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:63]: Neat. I like that more too.

File: contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol:100]: Also made it private

**Review by pinebit — APPROVED** [review]

**Review by KuphJr — APPROVED** [review]

**Pull request merged**
